packages/tf-inventory/src/stateFile.js

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable no-console */
+
+'use strict';
+
+const groupBy = require('lodash.groupby');
+const aws = require('@cumulus/common/aws');
+
+/**
+ * Get list of state files paths `bucket/key` from a table if the table
+ * contains state files
+ * @param {string} tableName - table name
+ * @returns {Promise<Array<string>>} - list of state file paths
+ */
+async function getStateFilesFromTable(tableName) {
+  try {
+    const tableInfo = await aws.dynamodb().describeTable({ TableName: tableName }).promise();
+
+    // Check that the table holds state files and actually has items
+    if (tableInfo.Table.AttributeDefinitions[0].AttributeName === 'LockID'
+        && tableInfo.Table.ItemCount > 0) {
+      let stateFiles = [];
+      const scanQueue = new aws.DynamoDbSearchQueue({ TableName: tableName });
+
+      let itemsComplete = false;
+
+      /* eslint-disable no-await-in-loop */
+      while (itemsComplete === false) {
+        await scanQueue.fetchItems();
+
+        itemsComplete = scanQueue.items[scanQueue.items.length - 1] === null;
+
+        if (itemsComplete) {
+          // pop the null item off
+          scanQueue.items.pop();
+        }
+
+        // Slice off the .md5 extension since this is listing the checksums
+        stateFiles = stateFiles.concat(scanQueue.items.map((i) => i.LockID.slice(0, -4)));
```

Comment:
Would it be a good idea to validate state file names here at some point rather before attempting to slice the `.md5`? I'm not sure since either way we'd be relying on an assumption (all tables with LockID in position [0] contain only statefiles OR state file names look like *something*).

Reply:
So you mean just validate it has an extension?

Reply:
Yeah. I'm not certain it needs to be changed to do that, just wanted to have the conversation.